### PR TITLE
Inclusion of a Tutorial on an Idealized Case & Parameter Description of Feature Detection

### PR DESCRIPTION
**Pull request by fsenf**
## Description
- an additional tutorial notebook on tracking of a 2d test blob
- two additional notebooks on the effect of feature detection parameters

Work mainly contributed by @snilsn and reviewed and updated by me.

## Checklist
* [x] Have you followed our guidelines in Section "Contributing" from README.md? 
* [x] Have you checked already existing notebooks to get an idea about our tutorial style?
* [x] Have you extensively used markdown elements to structure and explain your notebook?
* [x] Have you self-reviewed your code / notebook and corrected any misspellings?

* [ ] Have you uploaded your example data to [zenodo]( https://zenodo.org/ ) (see README) if needed?
   * does not apply
* [x] Have you checked if your notebook is running in the recommended standard conda enviroment [see here]((docs/RunningTutorials.md) and updates the [list of required python packages](./requirements.txt) if needed?
* [x] Have you carefully checked that sphinx-based rendering is still working correctly?
* [ ] Have you written descriptive commit messages? 
  * I am always bad on this aspect :-)
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [x] Have the newest changes been merged into this branch?

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -27,7 +39,7 @@ Welcome to a Collection of Tutorials for the Python Package `tobac`
 .. toctree::
    :maxdepth: 1 
    :hidden:
-   :caption: Tracking in Simulaitons
+   :caption: Advanced Tracking: Simulaitons
```
Comment (JuliaKukulies) [inline index.rst]:
```suggestion
   :caption: Advanced Tracking: Simulations
```

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-sphinx==3.5.4
+sphinx
```
Comment (JuliaKukulies) [inline requirements.txt:1]: Does the version not play any role anymore or why did you delete this?

File: tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb [hunk 3]
```diff
@@ -0,0 +1,3354 @@
...
+   "source": [
+    "# Idealized Case 1: Tracking of a Test Blob in 2D "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "be1be87f",
+   "metadata": {},
+   "source": [
+    "This tutorial shows the most important step of tracking with `tobac` using a idealized case:\n",
```
Comment (JuliaKukulies) [inline tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb]:
```suggestion
    "This tutorial shows the most important step of tracking with `tobac` using an idealized case:\n",
```

File: tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb [hunk 4]
```diff
@@ -0,0 +1,3354 @@
...
...jYjTK3dIOpjUi3xiOypX2KZ4vTmz76aK/RN552Ua9zdW7u9QmStIWpApt9J6k+w3MzMzG2ydXZDnQxHxUGWCpEtIncUfY9UO1b7USiP8C8CewGmS/o20kE8ALybFgt8GHNem+gFsWLzOz+ybD7ykKu8fSvJVltWJMs3MzMys0Mm47+oGeJG2QNIfgWd17szt08o84Y9LeiVwPPA2Vg7CfBQ4FfhMRDzWviquPHWD6fVueaN5p1ImETG7Tv6JnnL3hpuZmdn01eXBl5KeRopQGIiV21vpCSciFgIflPQhYA5pOfu/FrHS7fZI8Zrrcd6AVXuzH6mTj4q8nSjTzMzMzEgNQ03SLFR6WW+yMN7JOjcBJAk4jTTe8UsNVbLHWhmYuUIkf42IhzrUAAe4pXjdJrNvW1aN674F2CozxeC2xevNFfnaXaaZmZmZTRifZGuvE4G9gPdHxK1tL70DWmqES1pH0pHFIjt/lPSKIn1Okb5luyoYEfcC1wH7VTaEi6lpNgHOq8g+F5hNmve70v7AbRExr1NlmpmZmVmhWLa+3laEqyyMiNn1tslOJenzwGHAhyPizI6+rzZqOhyliLf5FWme7NuL1zUAIuJhSQeQGq0fabC8iSXuX1a87iBpDvBkRPysSPs4af7uH0g6DdgY+CJwDXBuRXEXAZcDp0vakDRC9gDStIl7Vp26E2WamZmZGXR6dhQAJB0LfAo4PCJO7vgJ26iVmPDPAc8AXk6aD7x6dOoFwM5NlHdu1c9HF693A5sCRMRlknYDjgF+CjwOnE+64CsWDoqIKBb9Oa7YZpOmD9wnIi6sPEknyjQzMzOzpNOrYko6CjgCOCIi2jk9dle00gjfDTglIq4veoar3Qkc2GhhEaEG8/0c+HkD+R4DDi22rpdpZmZmZnS0J1zSYaSO2/8CflG1QM+SiLihYydvk1Ya4XNIYShlxoHVW6uOmZmZmU0HbVgVs56JsXq7FVulFdEU/ayVRvhfgOfV2b8dDS5bb2ZmZmbTVAd7wiNix44V3iWtzI5yEXCIpGdW75D0ctKsIRdMtWJmZmZmNrgmnR1lyLXSCD8GWA7cQFo1M4ADJP0AuBK4nzTLiJmZmZkNq4j625BruhEeEX8BtidN5XcwacGjdwH7kqb8e3VEeBVJMzMzs2HW3cV6Bk6ry9b/GdhT0rrAC0kN8dvd+DYzMzMz
```
Comment (JuliaKukulies) [inline tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb]:
```suggestion
    "Another step after the feature detection and tracking is the segmentation of the data. This means we find the area surrounding our features belonging to the same cluster. Logically, we now need the already detected features as an additional input."
```

Comment (JuliaKukulies) [inline tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb]:
```suggestion
    "Another interesting property for real data are the lifetimes of our features. Tobac can also produce a histogram of this:"
```

File: tutorials/Basics/Methods-and-Parameters-for-Feature-Detection_Part_2.ipynb [hunk 5]
```diff
@@ -0,0 +1,1522 @@
...
+    "## Band-Pass Filter for Input Fields via Parameter `wavelength_filtering`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1bf56e78",
+   "metadata": {},
+   "source": [
+    "This parameter can be understood best, when looking at real instead of snythethic data. An example of usage is given [here](../Example_vorticity_tracking_model/Example_vorticity_tracking_model.ipynb)"
+   ]
```
Comment (JuliaKukulies) [inline tutorials/Basics/Methods-and-Parameters-for-Feature-Detection_Part_2.ipynb]: Maybe add this small section in `Methods-and-Parameters-for-Feature-Detection_Part_1` instead? It does probably not matter, but I felt it is more logically to mention the ability of wavelength filter together with the gaussian filter, because both input parameters apply some kind of filter on the input data.

File: tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb (hunk 4 above)
Comment (JuliaKukulies) [inline tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb]: _Just a small thought here:_ In all other example notebooks we introduce the different steps as 1.feature detection, 2.segmentation and 3. tracking. I know that the order actually does not matter since the segmentation is based on the feature detection output, so maybe we could clarify this here for users that the segmentation can be done before or after the linking. 

Also, one could actually use the output of the segmentation (`features_test` in your case) as input for the tracking with the advantage that information on the area (`ncells`) is added in the dataframe. One could also use the output of the tracking in the segmentation (`trajectories` in your case) with the advantage that  `mask` will contain only the features that are also linked to trajectories. Do you think these possibilities are worth mentioning here?

**Review by JuliaKukulies — COMMENTED** [review]
These three additional notebooks are VERY helpful and a great contribution in my opinion! Thanks a lot for this PR! I have only some small suggestions/questions and a few typo fixes before this can be merged.

File: tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb [hunk 6]
```diff
@@ -0,0 +1,3354 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "454d687c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dxy, dt = tobac.utils.get_spacings(test_data)"
+   ]
```
Comment (JuliaKukulies) [inline tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb:705]: should be `tobac.utils.general.get_spacings` for `v2.0-dev`

File: requirements.txt (hunk 2 above)
Comment (fsenf) [reply requirements.txt:1]: Yeah, the problem comes with older sphinx versions and their dependencies to other packages. With the fixed version v3.5.4 e.g., I get the following error after make html

...
ImportError: cannot import name 'environmentfilter' from 'jinja2' 

Using the newest sphinx version helps and no import errors appear.

I remember that we needed to fix the version (some time ago) because the at-this-time newest sphinx version had some small bugs.

PS: sorry for double posting ...

File: tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb (hunk 4 above)
Comment (fsenf) [reply tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb]: This comment refers to the image, right? Don't how to deal with it ...

Comment (fsenf) [reply tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb]: and the same: This comment refers to the image, right? Don't how to deal with it ...

Comment (fsenf) [reply tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb]: @JuliaKukulies : good point! I actually reordered Nils notebook such that segmentation comes later. The second segmentation is actually more an analysis step then something else. It does not impact tracking.

However, I did remember it wrong that the explanation in https://tobac.readthedocs.io/en/latest/ was also structured in this way. So now, I do have a tendency to shift it back again and to bring the second segmentation directly after the feature detection.

Comment (fsenf) [reply tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb]: I left it as it is ... but included some of your arguments, okay?

**Comment by fsenf** [comment]
@JuliaKukulies : The update should address most of your comments.

File: tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb (hunk 4 above)
Comment (JuliaKukulies) [reply tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb]: I am sorry, I have no idea why this image thing appears here. This was meant to be a direct suggestion for a super tiny typo: allready -> already, but I have realized that making direct changes in the notebook file somehow results in this when there is an image in the previous line.

Comment (JuliaKukulies) [reply tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb]: Same here, the only change I wanted to suggest was histgram -> histogram in l. 3297

Comment (JuliaKukulies) [reply tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb]: Yes, OK!

Comment (fsenf) [reply tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb]: OK, yes ... it is always a pain to review notebooks ... I corrected the spelling error. Update comes with the next commit ...

Comment (fsenf) [reply tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb]: Thanks for careful reading! This is also corrected in the next commit.

**Comment by fsenf** [comment]
@JuliaKukulies : all suggested correction should be included now! What do you think: Ready to merge?

**Review by JuliaKukulies — APPROVED** [review]
Great, thanks! Yes, please go ahead and merge!
